go/cmd/dolt/commands/log.go

```diff
@@ -198,3 +200,135 @@ func logCommits(ctx context.Context, dEnv *env.DoltEnv, cs *doltdb.CommitSpec, l
 
 	return 0
 }
+
+func tableExists(ctx context.Context, commit *doltdb.Commit, tableName string) (bool, error) {
+	rv, err := commit.GetRootValue()
+	if err != nil {
+		return false, err
+	}
+
+	_, ok, err := rv.GetTable(ctx, tableName)
+	if err != nil {
+		return false, err
+	}
+
+	return ok, nil
+}
+
+func logTableCommits(ctx context.Context, dEnv *env.DoltEnv, loggerFunc commitLoggerFunc, cs *doltdb.CommitSpec, tableName string, numLines int) int {
```

Comment:
In general, subroutines in cmd/ shouldn't return error codes, they should return errors that are handled by top-level cmd routines

So instead of logging errors and returning 1, just return an error and do the status code handling at the call site